File: 118. Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        ans.push_back({1});
        if (numRows == 1) 
        return ans;

        for (int i = 2; i <= numRows; i++) {
           vector<int> new_row(i);
           new_row[0]=1;
           new_row[i-1]=1;
           for (int j = 1; j<ans.back().size(); j++)
           {
               new_row[j]=ans.back()[j-1]+ans.back()[j];
           }
           ans.push_back(new_row);

        }   
        return ans;
    }
};

File: 118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>>ans = new ArrayList<>();

        if(numRows == 0)
        return ans;

        List<Integer>first = new ArrayList<>();
        first.add(1);
        ans.add(first);

        if(numRows == 1)
        return ans;

        for(int i=1;i<numRows;i++){
            List<Integer>prev = ans.get(i-1);
            List<Integer>row = new ArrayList<>();
            row.add(1);

            for(int j=0;j<i-1;j++){
                row.add(prev.get(j)+ prev.get(j+1));
            }

            row.add(1);
                ans.add(row);
        }
          return ans;
    }
  
}
